### PR TITLE
close session if HTTP response is not 200 or event is auth_revoked

### PR DESCRIPTION
**Pull request by dwatrous**
These changes make it possible to respond to auth failures and auth expirations

### CODE REVIEW

File: aiosseclient.py [hunk 1]
```diff
@@ -29,7 +32,10 @@ async def aiosseclient(url, last_id=None, **kwargs):
                     lines = []
                     continue
 
-                yield Event.parse(''.join(lines))
+                current_event = Event.parse(''.join(lines))
+                yield current_event
+                if current_event.event == "auth_revoked":
```
Comment (ebraminio) [inline aiosseclient.py:37]: First of all thanks for the filing the issue and the patch, doesn't this feel to be too specific? Maybe it's something that a client should do or maybe not. I'll apply the patch upon you response as if I trust you given you expertise on the field. Thanks 😊

File: aiosseclient.py [hunk 2]
```diff
@@ -20,6 +20,9 @@ async def aiosseclient(url, last_id=None, **kwargs):
                       sock_connect=None, sock_read=None)
     async with aiohttp.ClientSession(timeout=timeout) as session:
         response = await session.get(url, **kwargs)
+        if response.status != 200:
```
Comment (ebraminio) [inline aiosseclient.py:23]: Maybe something other than 200 also can be OK? Perhaps all the statuses starting 2 and maybe handling 3 also?

**Comment by dwatrous** [comment]
I wouldn't accept the PR as is. Based on the specification: https://html.spec.whatwg.org/multipage/server-sent-events.html, 

> Event stream requests can be redirected using HTTP 301 and 307 redirects as with normal HTTP requests. Clients will reconnect if the connection is closed; a client can be told to stop reconnecting using the HTTP 204 No Content response code.

What I do know is that a 401 should result in the session closing (and probably true for 4XX, 5XX and possibly others).

`auth_revoked` is an event type specified by Firebase Realtime Database, and probably isn't universal.

I'll have to think about how this could be modularized so that it would be easier to handle these cases for specific SSE implementations. For now I wanted to document that I had found a way to make this work for my case.

**Comment by ebraminio** [comment]
Fantastic, thank you very much. I've invited you to the repository as a collaborator. You can merge this PR whenever you like or let me merge, or leave it open as I understand the situation. Thanks 😊

**Comment by dwatrous** [comment]
I updated this to be parameterized with sensible defaults. Let me know what you think

**Comment by ebraminio** [comment]
Just made a release containing the changes https://pypi.org/manage/project/aiosseclient/releases/ using GitHub Actions CI, feel free to test and let me know if anything missing. It's even fine if you want to introduce a breaking API change to the just added API so feel free to test and tweak whatever you like or not. Thanks 😊

**Comment by ebraminio** [comment]
Also thanks to your work I've noticed https://github.com/rtfol/aiohttp-sse-client and https://github.com/JelleZijlstra/aiohttp-sse-client2 which started with inspiration from this project and as I believe we can learn from their experience now.

**Pull request merged**